### PR TITLE
feat: introduce personal access tokens for authorization

### PR DESCRIPTION
**Pull request by jamesgeorge007**
### Description

This PR includes the UI changes adding support for Personal Access Tokens (tokens associated with a user) providing a mechanism for authorization to access certain services and APIs the Hoppscotch platform offers. Currently, collections and environments from any team workspace (user having access to) can be accessed via PAT primarily for usage in the CLI, #4095.

The BE changes were added in #4079.

Closes HFE-519.

### Changes

- New tab addition in the `profile` page corresponding to personal access tokens.
- Add `AccessTokens` (parent component where other presentational components are rendered and include all business logic), `AccessTokensOverview`, `AccessTokensList`, and `AccessTokensGenerateModal` components compiling various flows wrt personal access tokens.
- Relevant `i18n` string entry additions.
- Adds a new parameter `includeTime` to the [shortDateTime](https://github.com/hoppscotch/hoppscotch/blob/main/packages/hoppscotch-common/src/helpers/utils/date.ts) helper function that defaults to `true` for an additional use case of conditionally including the time.

### Preview

https://github.com/hoppscotch/hoppscotch/assets/25279263/989bf51c-1a6f-45ef-a7e5-0fff8bb60552

### Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### Note to reviewers

- Currently, the network call to fetch the list of access tokens gets triggered while at the `Account` tab. A potential optimization that can be done here is ensuring the data fetch happens only while the respective tab is visited. The existing implementation ensures inactive tab contents are rendered (via the `render-inactive-tabs` prop for `HoppSmartTabs`) while landing on the initial `Account` tab and thought to revisit it separately as required.
 https://github.com/hoppscotch/hoppscotch/blob/acbea3aa9c41ca77f715c7d006a3acab9e7a7788/packages/hoppscotch-common/src/pages/profile.vue#L74-L78

- There's a layout shift while switching between the tabs under the profile page. A pre-existing issue that can be tackled separately.

- Since more subviews are getting added under the profile page, we could think in the direction of having dedicated routes for subviews where the common header element gets rendered at the parent `/profile` route with a slot for child routes `/profile/account`, etc later on.

### CODE REVIEW

File: packages/hoppscotch-common/src/components/accessTokens/List.vue [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    <div
+      v-for="{ id, label, lastUsedOn, expiresOn } in accessTokens"
+      :key="id"
+      class="flex items-center justify-between p-4 border rounded md:items-baseline md:flex-col md:gap-4 md:justify-normal border-divider"
+    >
+      <div class="w-full text-sm font-semibold truncate text-secondaryDark">
+        {{ label }}
+      </div>
+
+      <div class="flex items-center gap-x-4">
```
Comment (nivedin) [inline packages/hoppscotch-common/src/components/accessTokens/List.vue:12]:
```suggestion
    <div class="grid p-4 gap-4 sm:grid-cols-2 md:grid-cols-3 xl:grid-cols-4">
    <div
      v-for="{ id, label, lastUsedOn, expiresOn } in accessTokens"
      :key="id"
      class="flex items-center justify-between p-4 border rounded flex-col gap-4 border-divider"
    >
      <div class="w-full text-sm font-semibold truncate text-secondaryDark">
        {{ label }}
      </div>

      <div class="flex items-center w-full justify-between gap-x-4">
```

we can justify the Delete button , flex-col for smaller screens and set 3 cards for larger to xl screens..

File: packages/hoppscotch-common/src/components/accessTokens/index.vue [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+  />
+
+  <AccessTokensList
+    :access-tokens="accessTokens"
+    :has-more-tokens="hasMoreTokens"
+    :tokens-list-loading="tokensListLoading"
+    @delete-access-token="displayDeleteAccessTokenConfirmationModal"
+    @fetch-more-tokens="fetchAccessTokens"
+  />
+
```
Comment (nivedin) [inline packages/hoppscotch-common/src/components/accessTokens/index.vue:13]: should we add an empty state UI ?

File: packages/hoppscotch-common/src/pages/profile.vue [hunk 3]
```diff
@@ -177,14 +177,21 @@
...
+
             <HoppSmartTab :id="'teams'" :label="t('team.title')">
               <Teams :modal="false" class="p-4" />
             </HoppSmartTab>
+
+            <HoppSmartTab
+              id="tokens"
+              :label="t('access_tokens.tab_title')"
+              class="space-y-4"
+            >
```
Comment (nivedin) [inline packages/hoppscotch-common/src/pages/profile.vue:192]:
```suggestion
            <HoppSmartTab
              id="tokens"
              :label="t('access_tokens.tab_title')"
            >
```

**Review by nivedin — CHANGES_REQUESTED** [review]
please look into this comments

File: packages/hoppscotch-common/src/components/accessTokens/GenerateModal.vue [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+            {{ t("action.label") }}
+          </div>
+          <HoppSmartInput
+            v-model="accessTokenLabel"
+            placeholder=" "
+            class="floating-input"
+          />
+          <div class="text-secondaryLight">
+            {{ t("access_tokens.token_purpose") }}
+          </div>
```
[the remark this answers is not in the document]
Comment (jamesgeorge007) [reply packages/hoppscotch-common/src/components/accessTokens/GenerateModal.vue]: Yup, a placeholder will serve the purpose here; updated in https://github.com/hoppscotch/hoppscotch/pull/4094/commits/782b722303f034a461653e6ca9e3ce4d79492292.
cc @thetronjohnson

File: packages/hoppscotch-common/src/components/accessTokens/List.vue (hunk 1 above)
Comment (jamesgeorge007) [reply packages/hoppscotch-common/src/components/accessTokens/List.vue]: Good catch. Also, a 3-column layout didn't fit well for tablet devices falling in a breakpoint b/w `md` & `lg`. Hence, it is now applied only from the `lg` breakpoint.

> Before

  image

> After

  image

File: packages/hoppscotch-common/src/components/accessTokens/index.vue (hunk 2 above)
Comment (jamesgeorge007) [reply packages/hoppscotch-common/src/components/accessTokens/index.vue:14]: Had an initial thought that the overview section comprising the `Generate new token` action would serve the purpose. But, it's better to stay consistent with the pattern followed across list views where the error and empty states are displayed; updated in https://github.com/hoppscotch/hoppscotch/pull/4094/commits/782b722303f034a461653e6ca9e3ce4d79492292.

**Review by nivedin — APPROVED** [review]

**Review by AndrewBastin — APPROVED** [review]

**Pull request merged**